rmw_zenoh_cpp/src/detail/rmw_subscription_data.cpp

```diff
@@ -537,6 +537,7 @@ void SubscriptionData::add_new_message(
   // Since we added new data, trigger user callback and guard condition if they are available
   data_callback_mgr_.trigger_callback();
   if (wait_set_data_ != nullptr) {
+    std::lock_guard<std::mutex> wait_set_lock(wait_set_data_->condition_mutex);
```

Comment:
Should we lock above line 539 so that the check for `nullptr` is also protected by the lock?

Reply:
The lock resides inside the `wait_set_data` structure, so trying to take the lock before checking the validity of the pointer will very likely result in a `Segfault`.

Additionally the lock is taken in the same position as in [guard conditions](https://github.com/ros2/rmw_zenoh/blob/c7a80c285540b16730ec9fd7ccf5c9edf8b1900a/rmw_zenoh_cpp/src/detail/guard_condition.cpp#L40), [client data](https://github.com/ros2/rmw_zenoh/blob/c7a80c285540b16730ec9fd7ccf5c9edf8b1900a/rmw_zenoh_cpp/src/detail/rmw_client_data.cpp#L235), and [service data](https://github.com/ros2/rmw_zenoh/blob/c7a80c285540b16730ec9fd7ccf5c9edf8b1900a/rmw_zenoh_cpp/src/detail/rmw_service_data.cpp#L264).

Reply:
Oops I misread the code. Sorry for the noise!